### PR TITLE
chore(vscode): add vscode tasks and swift lang support

### PR DESCRIPTION
**Pull request by luqven**
# Description

Adds vscode tasks and apple [official](https://github.com/apple/swift-package-manager) lang extension for better IDE support.

> For more on what is being installed and how, read [here](https://nshipster.com/vscode/)

### CODE REVIEW

File: .swiftpm/xcode/package.xcworkspace/contents.xcworkspacedata [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Workspace
+   version = "1.0">
+   <FileRef
+      location = "self:">
+   </FileRef>
+</Workspace>
```
Comment (sherwinski) [inline .swiftpm/xcode/package.xcworkspace/contents.xcworkspacedata]: Just want to confirm that this is necessary and not some sort of addition specific to your local setup.

**Review by sherwinski — APPROVED** [review]
Any chance we can get a task in here that will install the `swift` CLI for users?

File: .swiftpm/xcode/package.xcworkspace/contents.xcworkspacedata (hunk 1 above)
Comment (luqven) [reply .swiftpm/xcode/package.xcworkspace/contents.xcworkspacedata]: 💯  you're right. Didn't realize the folder wasn't gitignored. My bad. Removing these.

**Comment by luqven** [comment]
> Any chance we can get a task in here that will install the `swift` CLI for users?

Unfortunately, on a mac, the only way to do this would be to add a script that installs xcode, the only way to get the swift CLI on a mac. Windows and Linux have different  dist. and implementations of the CLI. So I don't think it'd be easy to support this.

File: .vscode/extensions.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "recommendations": [
+    "kasik96.swift",
+    "vknabel.vscode-swift-development-environment"
+  ]
+}
```
Comment (luqven) [inline .vscode/extensions.json]: All this does is create a small prompt in vscode with recommended extensions.

File: .gitignore [hunk 3]
```diff
@@ -75,3 +75,5 @@ fastlane/test_output
 # End of https://www.gitignore.io/api/swift
 
 .DS_Store
+# Swift PM Workspace Specific Settings
+.swiftpm/xcode/package.xcworkspace
```
Comment (luqven) [inline .gitignore:79]: This was missing. It's the reason we were seeing `xcworkspace` files being checked in,

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "sde.languageServerMode": "sourcekit-lsp"
+}
```
Comment (sherwinski) [inline .vscode/settings.json]: Should this be checked in? Looks like it's related to an [extension](https://marketplace.visualstudio.com/items?itemName=vknabel.vscode-swift-development-environment)

**Pull request merged**